File: rust/src/models/update_product_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::product_fee_structure::ProductFeeStructure;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct UpdateProductRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub fee_structure: Option<ProductFeeStructure>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
}

impl UpdateProductRequest {
    pub fn new() -> Self {
        Self {
            description: None,
            fee_structure: None,
            name: None,
        }
    }
}

File: typescript/src/models/updateProductRequest.ts
// this file is @generated
import {
  type ProductFeeStructure,
  ProductFeeStructureSerializer,
} from "./productFeeStructure";

export interface UpdateProductRequest {
  description?: string | null;

  feeStructure?: ProductFeeStructure | null;

  name?: string | null;
}

export const UpdateProductRequestSerializer = {
  _fromJsonObject(object: any): UpdateProductRequest {
    return {
      description: object["description"],
      feeStructure:
        object["fee_structure"] != null
          ? ProductFeeStructureSerializer._fromJsonObject(object["fee_structure"])
          : undefined,
      name: object["name"],
    };
  },

  _toJsonObject(self: UpdateProductRequest): any {
    return {
      description: self.description,
      fee_structure:
        self.feeStructure != null
          ? ProductFeeStructureSerializer._toJsonObject(self.feeStructure)
          : undefined,
      name: self.name,
    };
  },
};
